### ISSUE TITLE
Incorrect timing report when asserts are not used before delays.

### ISSUE THREAD

**Issue by coreyfarrell** [tapjs/tapjs#587]
```js
'use strict'

const {promisify} = require('util')
const {test} = require('tap')

const delay = promisify(setTimeout)

test('assert after delay', async t => {
  await delay(1000)
  t.ok(true, 'test after 1 second delay')
})

test('assert before delay', async t => {
  t.ok(true, 'test before 1 second delay')
  await delay(1000)
})
```

This test file reports incorrect timing for `assert after delay` (tested on tap 14).
```
$ node test/timing.js |npx tap -
 PASS  assert after delay 1 OK 6ms
 PASS  assert before delay 1 OK 979ms

                         
  🌈 SUMMARY RESULTS 🌈  
                         

Suites:   2 passed, 2 of 2 completed
Asserts:  2 passed, of 2
Time:     1s
```

It seems that the reported time does not begin until the first assertion, but this can often be wrong for async testing.  Found this on graceful-fs testing test/avoid-memory-leak.js was reported as taking 28ms though it takes many seconds.

**Comment by isaacs** [comment]
Yeah, this looks like an issue with treport's timing when piping in.  I think in those cases it  should probably print the `# time=...` value instead of timing the actual test time.  When the "test" it's reporting is just parsing input, it's almost never going to match the actual test timing.  For example, even the `979ms`  is wrong -- it should be something a little above 1000ms.

**Comment by isaacs** [comment]
Fixed on 14.7.2
